rma/models/rma.py

```diff
@@ -670,21 +661,94 @@ def action_rma_send(self):
+        self, warehouse=None, scheduled_date=None, group=None
+    ):
+        self.ensure_one()
+        group = group or self.procurement_group_id
+        if not group:
+            group = self.env["procurement.group"].create(
+                self._prepare_procurement_group_values()
+            )
+        return {
+            "company_id": self.company_id,
+            "group_id": group,
+            "date_planned": scheduled_date or fields.Datetime.now(),
+            "warehouse_id": warehouse or self.warehouse_id,
+            "partner_id": group.partner_id.id,
+            "priority": self.priority,
+        }
+
+    def _prepare_reception_procurement_values(self, group=None):
+        values = self._prepare_procurement_values(group=group)
+        values.update(
+            {
+                "rma_receiver_ids": [(6, 0, self.ids)],
+            }
+        )
+        if self.move_id:
+            values.update(
+                {
+                    "origin_returned_move_id": self.move_id.id,
+                }
+            )
+        return values
+
+    def _prepare_reception_procurements(self):
+        procurements = []
+        group_model = self.env["procurement.group"]
+        for rma in self:
+            if not rma._product_is_storable():
+                continue
+            group = rma.procurement_group_id
+            if not group:
+                group = group_model.create(rma._prepare_procurement_group_values())
+            product = rma.product_id
+            values = rma._prepare_reception_procurement_values(group)
+            values.update({"route_ids": rma.warehouse_id.rma_in_route_id})
+            procurements.append(
+                group_model.Procurement(
+                    product,
+                    rma.product_uom_qty,
+                    rma.product_uom,
+                    rma.location_id,
+                    product.display_name,
+                    group.name,
+                    rma.company_id,
+                    values,
+                )
+            )
+        return procurements
+
     def action_confirm(self):
         """Invoked when 'Confirm' button in rma form view is clicked."""
-        self.ensure_one()
         self._ensure_required_fields()
-        if self.state == "draft":
-            if self.picking_id:
-                reception_move = self._create_receptions_from_picking()
-            else:
-                reception_move = self._create_receptions_from_product()
-            self.write({"reception_move_id": reception_move.id, "state": "confirmed"})
-            self._add_message_subscribe_partner()
-            self._send_confirmation_email()
+        self = self.filtered(lambda rma: rma.state == "draft")
+        if not self:
+            return
+        procurements = self._prepare_reception_procurements()
+        if procurements:
+            self.env["procurement.group"].run(procurements)
+        self.reception_move_id.picking_id.action_assign()
```

Comment:
Where is `reception_move_id` written?

Reply:
This is done with the `rma_receiver_ids` key within _values_ in the procurement